x-pack/plugins/reporting/server/lib/tasks/execute_report.ts

```diff
@@ -148,11 +171,20 @@ export class ExecuteReportTask implements ReportingTask {
     const m = moment();
 
     // check if job has exceeded the configured maxAttempts
-    const maxAttempts = this.config.capture.maxAttempts;
+    const maxAttempts = this.getMaxAttempts();
     if (report.attempts >= maxAttempts) {
-      const err = new QueueTimeoutError(
-        `Max attempts reached (${maxAttempts}). Queue timeout reached.`
-      );
+      let err: ReportingError;
+      if (report.error && isExecutionError(report.error)) {
+        // We have an error stored from a previous attempts, so we'll use that
+        // error to fail the job and return it to the user.
+        const { error } = report;
+        err = mapToReportingError(error);
+        err.stack = error.stack;
+      } else {
+        err = new QueueTimeoutError(
```

Comment:
Actually this condition is a fallback mechanism. We should _always_ have an error on the report once we enter the condition `if (report.attempts >= maxAttempts)`.

We could remove the `if/else` statement but for the sake of robustness I would leave it. WDYT?

Reply:
This is a nit, and I don't want to hold anything up for this. My comment was about: what if `report.error` is truthy, but `isExecutionError(report.error)` is false? I thought to just add a `errorLogger(report.error)` under the `else`. It probably will never log anything, but there could be a day when it ends up being useful in log investigation. It might only be possible through an unintentional mistake in the code.

The rest looks good: we would still want to label the error as a `QueueTimeoutError`.

Reply:
I see what you mean. I added an error logger in https://github.com/elastic/kibana/pull/173775/commits/4c83886cfd9bb8e5045cd587f3e05472a4fa8a5d